benchmarks/setup/utils.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+GO_OS=${GO_OS:-"linux"}
+
+function detect_os {
+    # Detect the OS name
+    case "$(uname -s)" in
+      Darwin)
+        host_os=darwin
+        ;;
+      Linux)
+        host_os=linux
+        ;;
+      *)
+        echo "Unsupported host OS. Must be Linux or Mac OS X." >&2
+        exit 1
+        ;;
+    esac
+
+   GO_OS="${host_os}"
+}
+
+
+function sed_inplace() {
```

Comment:
why define this special function? Why not just turn this util script into `sed-inplace.sh`? And everytime you need it you just call this script?

Reply:
Hm good question. I started with it because I thought I move more out but then it was just this. Idk I can also replace it, but does it matter? Do you have a preference for it?

Reply:
Not really, I think the only advantage on the calling side is not having to "import" the function. That's negligible.